### PR TITLE
extractData2() for trend_GADSdat

### LINKED ISSUE #29: extractData2() for trend_GADSdat objects
Implement S3 method of `extractData2()` for `trend_GADSdat` objects.

### PR DESCRIPTION
**Pull request by beckerbenj**
Implements `extractData2()` S3 method for `trend_GADSdat` objects. Closes #29. 

Needed for tomorrows BT22 workshop. Sorry for the urgency!

### CODE REVIEW

File: tests/testthat/test_extractData2.R [hunk 1]
```diff
@@ -198,6 +198,42 @@ test_that("extractData2 with some variables labels applied to (convertVariables
...
+  out <- extractData2(trend_gads)
+  expect_equal(dim(out), c(6, 5))
+  expect_equal(names(out), c("ID1", "V1", "V2", "V3", "year"))
+  comp <- c(rep(2012, 3), c(rep(2018, 3)))
+  attr(comp, "label") <- "Trendvariable, indicating the year of the assessment"
+  expect_equal(out$year, comp)
+
+  ## convertVariables if some variables are not in both GADS
+  out2 <- extractData(trend_gads, convertVariables = "V3")
+  expect_equal(out, out2)
```
Comment (franikowsp) [inline tests/testthat/test_extractData2.R]: Nit (?): Yields the same result if `convertVariables = "V3"` is omitted (V3 seems to be treated equally across `extractData()` and `extractData2()`?

Comment (franikowsp) [inline tests/testthat/test_extractData2.R]: LGTM!

File: tests/testthat/test_extractData2.R [hunk 2]
```diff
@@ -198,6 +198,42 @@ test_that("extractData2 with some variables labels applied to (convertVariables
...
+  fp1 <- system.file("extdata", "trend_gads_2020.db", package = "eatGADS")
+  fp2 <- system.file("extdata", "trend_gads_2015.db", package = "eatGADS")
+  fp3 <- system.file("extdata", "trend_gads_2010.db", package = "eatGADS")
+  s <- capture_output(gads_3mp <- getTrendGADS(filePaths = c(fp1, fp2, fp3), years = c(2020, 2015, 2010), fast = FALSE, verbose = FALSE))
+
+  out <- extractData2(gads_3mp)
+  expect_equal(dim(out), c(180, 10))
+  expect_equal(dim(out), c(180, 10))
+  expect_equal(names(out), c("idstud", "gender", "dimension", "imp", "score", "traitLevel", "failMin", "passReg", "passOpt", "year"))
+  expect_equal(out$dimension[1:2], c(1, 2))
```
Comment (franikowsp) [inline tests/testthat/test_extractData2.R:247]: Nit: Would it be better to check for the whole vector (something like `rep(c(1, 2), 90)`)?

File: tests/testthat/test_extractData2.R [hunk 3]
```diff
@@ -198,6 +198,42 @@ test_that("extractData2 with some variables labels applied to (convertVariables
...
+  expect_equal(as.numeric(out$year), c(rep(2020, 60), rep(2015, 60), rep(2010, 60)))
+
+  out2 <- extractData2(gads_3mp, labels2character = "dimension")
+  expect_equal(out2$dimension[1:2], c("listening", "reading"))
+
+  out3 <- extractData2(gads_3mp, labels2factor = "dimension")
+  expect_equal(out3$dimension[1:2], factor(c("listening", "reading")))
```
Comment (franikowsp) [inline tests/testthat/test_extractData2.R:254]: Nit: Same for these (check for the whole vector)... But I guess, that is more economic (I have not worked with `eatPrep`, yet).

File: R/extractData2.R [hunk 4]
```diff
@@ -92,9 +92,19 @@ extractData2.GADSdat <- function(GADSdat, convertMiss = TRUE, labels2character =
...
-extractData2.trend_GADSdat <- function(GADSdat, convertMiss = TRUE, labels2character = namesGADS(GADSdat), labels2factor = NULL,
+extractData2.trend_GADSdat <- function(GADSdat, convertMiss = TRUE, labels2character = NULL, labels2factor = NULL,
                                        labels2ordered = NULL, dropPartialLabels = TRUE) {
-  stop("extractData2 has not been implemented for 'trend_GADSdat' objects.")
+  check_trend_GADSdat(GADSdat)
+
+  dat_list <- lapply(names(GADSdat$datList), function(nam) {
+    gads <- extractGADSdat(all_GADSdat = GADSdat, name = nam)
+    dat <- extractData2(gads, convertMiss = convertMiss, labels2character = labels2character, labels2factor = labels2factor,
+                       labels2ordered = labels2ordered, dropPartialLabels = dropPartialLabels)
```
Comment (franikowsp) [inline R/extractData2.R]: Nit: For the sake of readability, I would use a new line for every argument (also holds for the function definition).

``` r
dat <- extractData2(gads, 
                    convertMiss = convertMiss, 
                    labels2character = labels2character, 
                    labels2factor = labels2factor,
                    labels2ordered = labels2ordered, 
                    dropPartialLabels = dropPartialLabels)                     
```

Comment (franikowsp) [reply R/extractData2.R]: The rest: LGTM!

File: vignettes/getGADS.Rmd
Comment (franikowsp) [inline vignettes/getGADS.Rmd:1]: LGTM!

**Review by franikowsp — APPROVED** [review]
Looks all good to me - therefore, only nits again! 

I hope everything works fine for the workshop tomorrow!

File: tests/testthat/test_extractData2.R (hunk 1 above)
Comment (beckerbenj) [reply tests/testthat/test_extractData2.R]: Fixed (should have been extractData2). Modified the test, so `V3` has actually value labels that can be applied.

File: tests/testthat/test_extractData2.R (hunk 2 above)
Comment (beckerbenj) [reply tests/testthat/test_extractData2.R:247]: Fixed!

File: tests/testthat/test_extractData2.R (hunk 3 above)
Comment (beckerbenj) [reply tests/testthat/test_extractData2.R:254]: Fixed both!

File: R/extractData2.R (hunk 4 above)
Comment (beckerbenj) [reply R/extractData2.R]: Fixed! (although I am not sure whether you meant something else with function definition?)

Comment (franikowsp) [reply R/extractData2.R]: Sorry, that was imprecise: I meant the function assignment (but you already addressed it with your commit):

``` r
fun_name <- function(arg1 = NULL,
                     arg2 = NULL,
                     arg3 = NULL) { ... }
```

**Pull request merged**